apps/web/src/components/Basenames/RegistrationForm/index.tsx

```diff
@@ -38,9 +47,20 @@ function formatUsdPrice(price: bigint, ethUsdPrice: number) {
 }
 
 export default function RegistrationForm() {
+  const { isConnected, chain: connectedChain, address } = useAccount();
+  const chains = useChains();
   const { openConnectModal } = useConnectModal();
   const { logEventWithContext } = useAnalytics();
   const { basenameChain } = useBasenameChain();
+  const { switchChain } = useSwitchChain();
+  const switchToIntendedNetwork = useCallback(
```

Comment:
This is the exact same as the other `switchToIntendedNetwork` and if so could U extract it?